rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -526,9 +525,10 @@ def _combine_sparse_dense_features(
             else:
                 dense_features.append(f)
 
-        output = tf.concat(dense_features, axis=-1) * mask
-        # apply mean to convert sequence to sentence features
-        output = tf.reduce_sum(output, axis=1) / tf.reduce_sum(mask, axis=1)
+        # extract the CLS token feature vector
+        dense_features = [f[:, -1, :] for f in dense_features]
```

Comment:
I think we need a mask to extract the last token due to different sequence lengths. Let's use the following:
```
sequence_lengths = tf.cast(tf.reduce_sum(mask[:, :, 0], 1), tf.int32)
last_index = tf.maximum(tf.constant(0, dtype=sequence_lengths.dtype), sequence_lengths - 1)
idxs = tf.stack([tf.range(tf.shape(last_index)[0]), last_index], axis=1)
cls = tf.gather_nd(text_transformed, idxs)
```

Reply:
true, but should not be needed anymore once we extract the features already in `create_session_data`

Reply:
oh, then yes, I didn't see that